ablateLibrary/boundarySolver/boundarySolver.cpp

```diff
@@ -312,3 +321,63 @@ PetscErrorCode ablate::boundarySolver::BoundarySolver::ComputeRHSFunction(PetscR
     VecRestoreArrayRead(cellGeomVec, &cellGeomArray) >> checkError;
     PetscFunctionReturn(0);
 }
+void ablate::boundarySolver::BoundarySolver::InsertFieldFunctions(const std::vector<std::shared_ptr<mathFunctions::FieldFunction>>& fieldFunctions, PetscReal time) {
```

Comment:
This also seems fine, but you might be able to do the same thing with DMProjectFunction(). If you can't, I would be interested, because it means I might be able to extend it in some way.

Reply:
I'm projecting to the cell face.  For cells with multiple faces I'm currently representing the face as the average centroid location and the sum of the normal areas.